src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs

```diff
@@ -155,9 +155,10 @@ public override int Run(InterpretedFrame frame)
             {
                 if (frame.Peek() == null)
                 {
-                    frame.Pop();
-                    throw new InvalidOperationException();
+                    // Trigger InvalidOperationException with same localised method as if we'd called the Value getter.
```

Comment:
Nit: localised => localized... I realize the former is a valid British spelling, but the codebase generally uses the American form